src/pt/lycantoons/src/eu/kanade/tachiyomi/extension/pt/lycantoons/LycanToons.kt

```diff
@@ -0,0 +1,190 @@
+            status = filters.valueOrEmpty<StatusFilter>(),
+            tags = filters.selectedTags(),
+        )
+
+        val body = json.encodeToString(payload).toRequestBody(JSON_MEDIA_TYPE)
+        return POST("$baseUrl/api/series", headers, body)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage =
+        response.parseAs<SearchResponse>().series.toMangasPage()
+
+    override fun getFilterList(): FilterList = LycanToonsFilters.get()
+
+    // =====================Details=====================
+
+    override fun mangaDetailsRequest(manga: SManga): Request = seriesRequest(manga.slug())
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        val result = response.parseAs<SeriesDto>()
+        return result.toSManga().apply { initialized = true }
+    }
+
+    // =====================Chapters=====================
+
+    override fun chapterListRequest(manga: SManga): Request = seriesRequest(manga.slug())
+
+    override fun chapterListParse(response: Response): List<SChapter> =
+        response.parseAs<SeriesDto>().let { series ->
+            series.capitulos.orEmpty()
+                .map { it.toSChapter(series.slug) }
+                .sortedByDescending { it.chapter_number }
+        }
+
+    // =====================Pages========================
+
+    override fun pageListRequest(chapter: SChapter): Request = GET(
+        "$baseUrl${chapter.url}",
+        pageHeaders,
+    )
+
+    override fun pageListParse(response: Response): List<Page> {
+        val html = response.body.string()
+        val (slug, chapterNumber) = response.extractSlugAndChapter()
+
+        val pageCount = pageCountFromHtml(html, chapterNumber)
+            ?.takeIf { it > 0 }
+            ?: error("Quantidade de páginas não encontrada no HTML para capítulo '$chapterNumber'")
+
+        val chapterPath = "$cdnUrl/$slug/$chapterNumber"
+        return List(pageCount) { index ->
+            val imageUrl = "$chapterPath/page-$index.jpg"
+            Page(index, imageUrl = imageUrl)
+        }
+    }
+
+    override fun imageUrlParse(response: Response): String = throw UnsupportedOperationException()
+
+    // =====================Utils=====================
+
+    private fun Response.extractSlugAndChapter(): Pair<String, String> {
+        val segments = request.url.pathSegments
+        val slug = segments.getOrNull(1)?.trim('/')
+            ?: throw IllegalStateException("Slug da série não encontrado na URL")
+        val chapterNumber = segments.getOrNull(2)?.trim('/')
+            ?: throw IllegalStateException("Número do capítulo não encontrado na URL")
+        return slug to chapterNumber
+    }
+
+    private fun pageCountFromHtml(html: String, chapterNumber: String): Int? = try {
+        val jsonStr = getChapterDataJson(html)
+        val cleanJsonStr = jsonStr.replace(Regex("\\\\\""), "\"")
+        val jsonObj = json.parseToJsonElement(cleanJsonStr).jsonObject
```

Comment:
Maybe reference implementation in: https://github.com/keiyoushi/extensions-source/blob/a229f392e13067105dab0b45ddb25b46096a2d18/lib-multisrc/greenshit/src/eu/kanade/tachiyomi/multisrc/greenshit/GreenShit.kt#L231-L236